[6.8] A problem occurred evaluating project ':benchmarks'.

`A problem occurred evaluating project ':benchmarks'.`

Build 20200427115226-68434CDB
Log https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+matrix-java-periodic/ES_RUNTIME_JAVA=openjdk15,nodes=general-purpose/656/console
Build Scans https://gradle-enterprise.elastic.co/s/mmxpvcs6ulojy

The following can be found in logs:
```
Caused by: java.lang.IllegalArgumentException: Could not determine java version from ''.
	at org.gradle.api.JavaVersion.assertTrue(JavaVersion.java:227)
	at org.gradle.api.JavaVersion.findFirstNonVersionCharIndex(JavaVersion.java:254)
	at org.gradle.api.JavaVersion.toVersion(JavaVersion.java:80)
	at org.gradle.api.JavaVersion$toVersion$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)
	at org.elasticsearch.gradle.BuildPlugin.globalBuildInfo(BuildPlugin.groovy:155)
	at org.elasticsearch.gradle.BuildPlugin$globalBuildInfo$3.callStatic(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:55)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:196)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:208)
	at org.elasticsearch.gradle.BuildPlugin.apply(BuildPlugin.groovy:102)
	at org.elasticsearch.gradle.BuildPlugin.apply(BuildPlugin.groovy)
	at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:42)
	at org.gradle.api.internal.plugins.RuleBasedPluginTarget.applyImperative(RuleBasedPluginTarget.java:50)
	at org.gradle.api.internal.plugins.DefaultPluginManager.addPlugin(DefaultPluginManager.java:177)
	at org.gradle.api.internal.plugins.DefaultPluginManager.access$300(DefaultPluginManager.java:51)
	at org.gradle.api.internal.plugins.DefaultPluginManager$AddPluginBuildOperation.run(DefaultPluginManager.java:267)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:155)
	at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:152)
	at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:48)
	at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:152)
```

As @droberts195 pointed out it may be due to using Java 15 in the old (6.8) build.

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
@@ -157,7 +157,7 @@ class BuildPlugin implements Plugin<Project> {
                 runtimeJavaVersionEnum = JavaVersion.toVersion(findJavaSpecificationVersion(project, runtimeJavaHome))
             }
 
-            String inFipsJvmScript = 'print(java.security.Security.getProviders()[0].name.toLowerCase().contains("fips"));'
+            String inFipsJvmScript = 'java.security.Security.getProviders()[0].name.toLowerCase().contains("fips")'
             boolean inFipsJvm = Boolean.parseBoolean(runScript(project, runtimeJavaHome, inFipsJvmScript))
 
             // Build debugging info

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
@@ -23,6 +23,7 @@ import com.github.jengelman.gradle.plugins.shadow.ShadowPlugin
 import org.apache.commons.io.IOUtils
 import org.apache.tools.ant.taskdefs.condition.Os
 import org.elasticsearch.gradle.precommit.PrecommitTasks
+import org.gradle.api.Action
 import org.gradle.api.GradleException
 import org.gradle.api.InvalidUserDataException
 import org.gradle.api.JavaVersion
@@ -53,6 +54,7 @@ import org.gradle.api.tasks.javadoc.Javadoc
 import org.gradle.internal.jvm.Jvm
 import org.gradle.process.ExecResult
 import org.gradle.process.ExecSpec
+import org.gradle.process.internal.ExecException
 import org.gradle.util.GradleVersion
 
 import java.nio.charset.StandardCharsets
@@ -156,7 +158,7 @@ class BuildPlugin implements Plugin<Project> {
             }
 
             String inFipsJvmScript = 'print(java.security.Security.getProviders()[0].name.toLowerCase().contains("fips"));'
-            boolean inFipsJvm = Boolean.parseBoolean(runJavaAsScript(project, runtimeJavaHome, inFipsJvmScript))
+            boolean inFipsJvm = Boolean.parseBoolean(runScript(project, runtimeJavaHome, inFipsJvmScript))
 
             // Build debugging info
             println '======================================='
@@ -446,46 +448,52 @@ class BuildPlugin implements Plugin<Project> {
 
     /** Finds printable java version of the given JAVA_HOME */
     private static String findJavaVersionDetails(Project project, String javaHome) {
-        String versionInfoScript = 'print(' +
-            'java.lang.System.getProperty("java.vendor.version", java.lang.System.getProperty("java.vendor")) + " " + ' +
+        String versionInfoScript = 'java.lang.System.getProperty("java.vendor.version", java.lang.System.getProperty("java.vendor")) + " " + ' +
             'java.lang.System.getProperty("java.version") + " [" +' +
             'java.lang.System.getProperty("java.vm.name") + " " + ' +
-            'java.lang.System.getProperty("java.vm.version") + "]");'
-        return runJavaAsScript(project, javaHome, versionInfoScript).trim()
+            'java.lang.System.getProperty("java.vm.version") + "]"'
+        return runScript(project, javaHome, versionInfoScript).trim()
     }
 
     /** Finds the parsable java specification version */
     private static String findJavaSpecificationVersion(Project project, String javaHome) {
-        String versionScript = 'print(java.lang.System.getProperty("java.specification.version"));'
-        return runJavaAsScript(project, javaHome, versionScript)
-    }
-
-    private static String findJavaVendor(Project project, String javaHome) {
-        String vendorScript = 'print(java.lang.System.getProperty("java.vendor.version", System.getProperty("java.vendor"));'
-        return runJavaAsScript(project, javaHome, vendorScript)
-    }
-
-    /** Finds the parsable java specification version */
-    private static String findJavaVersion(Project project, String javaHome) {
-        String versionScript = 'print(java.lang.System.getProperty("java.version"));'
-        return runJavaAsScript(project, javaHome, versionScript)
+        String versionScript = 'java.lang.System.getProperty("java.specification.version")'
+        return runScript(project, javaHome, versionScript)
     }
 
     /** Runs the given javascript using jjs from the jdk, and returns the output */
-    private static String runJavaAsScript(Project project, String javaHome, String script) {
-        ByteArrayOutputStream stdout = new ByteArrayOutputStream()
-        ByteArrayOutputStream stderr = new ByteArrayOutputStream()
+    private static String runScript(Project project, String javaHome, String script) {
         if (Os.isFamily(Os.FAMILY_WINDOWS)) {
             // gradle/groovy does not properly escape the double quote for windows
             script = script.replace('"', '\\"')
         }
-        File jrunscriptPath = new File(javaHome, 'bin/jrunscript')
-        ExecResult result = project.exec {
-            executable = jrunscriptPath
-            args '-e', script
-            standardOutput = stdout
-            errorOutput = stderr
-            ignoreExitValue = true
+
+        try {
+            File jrunscriptPath = new File(javaHome, 'bin/jrunscript')
+            return exec(project, false) { spec ->
+                spec.executable = jrunscriptPath
+                spec.args '-e', "print($script);"
+            }
+        } catch (ExecException ex) {
+            // if jrunscript fails, let's try jshell
+            ByteArrayInputStream input = new ByteArrayInputStream("System.err.print(${script});\n".getBytes('UTF-8'))
+            File jshellPath = new File(javaHome, 'bin/jshell')
+            return exec(project, true) { spec ->
+                spec.executable = jshellPath
+                spec.args '-s'
+                spec.standardInput = input
+            }
+        }
+    }
+
+    private static String exec(Project project, boolean captureStdErr, Action<? super ExecSpec> spec) {
+        ByteArrayOutputStream stdout = new ByteArrayOutputStream()
+        ByteArrayOutputStream stderr = new ByteArrayOutputStream()
+        ExecResult result = project.exec { ExecSpec s ->
+            spec.execute(s)
+            s.standardOutput = stdout
+            s.errorOutput = stderr
+            s.ignoreExitValue = true
         }
         if (result.exitValue != 0) {
             project.logger.error("STDOUT:")
@@ -493,8 +501,9 @@ class BuildPlugin implements Plugin<Project> {
             project.logger.error("STDERR:")
             stderr.toString('UTF-8').eachLine { line -> project.logger.error(line) }
             result.rethrowFailure()
+            result.assertNormalExitValue() // assert exit value in case the failure cause is null
         }
-        return stdout.toString('UTF-8').trim()
+        return (captureStdErr ? stderr : stdout).toString('UTF-8').trim()
     }
 
     /** Return the configuration name used for finding transitive deps of the given dependency. */

--- a/x-pack/plugin/sql/qa/security/with-ssl/build.gradle
+++ b/x-pack/plugin/sql/qa/security/with-ssl/build.gradle
@@ -205,10 +205,7 @@ integTestCluster {
     return tmpFile.exists()
   }
 }
-Closure notRunningFips = {
-  Boolean.parseBoolean(BuildPlugin.runJavaAsScript(project, project.runtimeJavaHome,
-          'print(java.security.Security.getProviders()[0].name.toLowerCase().contains("fips"));')) == false
-}
+Closure notRunningFips = { rootProject.ext.inFipsJvm == false }
 
 // Do not attempt to form a cluster in a FIPS JVM, as doing so with a JKS keystore will fail.
 // TODO Revisit this when SQL CLI client can handle key/certificate instead of only Keystores.

